app/scripts/views/settings/settings-file-view.js

```diff
@@ -512,10 +512,22 @@ const SettingsFileView = Backbone.View.extend({
                 this.backupInProgress = false;
                 backupButton.text(Locale.setFileBackupNow);
                 if (err) {
+                    let title = ''
+                    let description = ''
+                    if (err.hasOwnProperty('code') && err.code === 'EISDIR') {
```

Comment:
The error code is node-specific: it shouldn't be here. Please move to storage implementation and return as a property of error from there.

Reply:
You mean like modify the returned error object in i.e. StorageFile#save to include .isDir=true when code === EISDIR and something like that for the other possible storage providers?

Reply:
Yes exactly. To move node fs dependencies into fs provider.

Reply:
✅